misk/src/main/kotlin/misk/client/HttpClientsConfig.kt

```diff
@@ -1,34 +1,96 @@
 package misk.client
 
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize
+import com.fasterxml.jackson.databind.type.TypeFactory
 import misk.config.Config
 import misk.security.ssl.CertStoreConfig
 import misk.security.ssl.TrustStoreConfig
+import java.io.File
 import java.time.Duration
 
+data class BackwardsCompatibleEndpointConfig(
+    //Common fields
+  val url: String? = null,
+  val envoy: HttpClientEnvoyConfig? = null,
+
+    //Legacy fields
```

Comment:
These are all deprecated? Maybe worth noting that in a comment

Reply:
Tough question. This is a backwards compatibility layer, so the layer itself is not deprecated (yet) and will never be - it will just be deleted when we do not care about backwards compatibility anymore. I think I won't mark them `@Deprecated`, as to me `@Deprecated` sounds as "do not use", while nobody will be using any of these classes anyway. I would have made them private/internal, but Jackson does not like that idea.